web/src/engine/websites/ReadComicOnlineLI.ts

```diff
@@ -0,0 +1,40 @@
+import { Tags } from '../Tags';
+import icon from './ReadComicOnlineLI.webp';
+import { DecoratableMangaScraper } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+
+function ChapterExtractor(anchor: HTMLAnchorElement) {
+    const link = new URL(anchor.href);
+    link.searchParams.set('readType', '1');
+    link.searchParams.set('quality', 'hq');
+    return {
+        id: link.pathname + link.search,
+        title: anchor.text.trim()
+    };
+}
+
+function MangInfoExtractor(element: HTMLDivElement) { //needed because titles are truncated in link
+    const anchor = element.querySelector<HTMLAnchorElement>('a');
+    const dom = new DOMParser().parseFromString(element.getAttribute('title'), 'text/html');
+    return {
+        id: anchor.pathname,
+        title: dom.querySelector('p.title').textContent.trim()
+    };
+}
+
+@Common.MangaCSS(/^{origin}\/Comic\/[^/]+$/, 'div.barContent a.bigChar')
+@Common.MangasMultiPageCSS('/ComicList?page={page}', 'div.list-comic div.item, div.item-list div.group div.col.info', 1, 1, 0, MangInfoExtractor)
+@Common.ChaptersSinglePageCSS('div.episodeList table.listing tr td:first-of-type a, div.section ul.list li a', ChapterExtractor)
+@Common.PagesSinglePageJS('lstImages', 2500) //may trigger a captcha request
```

Comment:
What about the parameters `readType=1` and `quality=hq`, are these no longer necessary?

Reply:
They are necessary when getting pages. Hence the ChapterExtractor :

![image](https://github.com/manga-download/haruneko/assets/10671518/89d6ed75-52fe-47ae-86e2-297e972ed92a)

Reply:
I see. Not a big fan of adding redundant information to the `Identifier`, but since the chapter `Identifier` isn't that much relevant in any other places, this seems to be a good solution.

Reply:
Yeah, i'd rather do that than overriding a method,  Its more elegant.